### PR TITLE
DEV: Introduce <DPageHeader /> and <DPageSubheader /> components

### PR DESCRIPTION
**Pull request by martin-brennan**
This converts the `<AdminPageHeader />` component and the
`<AdminPageSubheader />` components into new components
that can be used outside of admin, and updates the CSS classes.
Also introduces a `<DPageActionButton />` component and child
components for the header action buttons.

I have to keep the old admin-only components around for
now until plugins are updated, then we can remove it,
and remove the re-exports that are done within
admin-page-action-button.gjs

For reference it's this part of the page:

image

### CODE REVIEW

**Review by ZogStriP — APPROVED** [review]

File: app/assets/javascripts/admin/addon/templates/config-about.hbs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<AdminPageHeader
+<DPageHeader
   @titleLabel="admin.config_areas.about.header"
   @descriptionLabelTranslated={{i18n
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/addon/templates/config-about.hbs:3]: @martin-brennan I think we shouldn't do this kind of stuff.  We should always consider the label we give here is translated. I get that we can think that passing a key is nice but it just creates a weird API, where you have support foo/fooTranslated for every similar properties.

I strongly recommend:

```gjs
@titleLabel={{i18n "admin.config_areas.about.header"}}
```

File: app/assets/javascripts/admin/addon/templates/config-look-and-feel.hbs [hunk 2]
```diff
@@ -1,9 +1,10 @@
-<AdminPageHeader
+<DPageHeader
   @titleLabel="admin.config_areas.look_and_feel.title"
   @descriptionLabel="admin.config_areas.look_and_feel.description"
   @learnMoreUrl="https://meta.discourse.org/t/beginners-guide-to-using-discourse-themes/91966"
 >
   <:breadcrumbs>
+    <DBreadcrumbsItem @path="/admin" @label={{i18n "admin_title"}} />
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/addon/templates/config-look-and-feel.hbs:7]: that's a detail but in ember it's generally called route not path

File: app/assets/javascripts/discourse/app/components/d-page-action-button.gjs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    ...attributes
+    @action={{@action}}
+    @route={{@route}}
+    @routeModels={{@routeModels}}
+    @label={{@label}}
+    @title={{@title}}
+    @icon={{@icon}}
+    @isLoading={{@isLoading}}
+  />
+</template>;
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/d-page-action-button.gjs:110]: So we could just reuse the same button then?

File: app/assets/javascripts/discourse/app/components/d-page-header.gjs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                @identifier="d-page-header-mobile-actions"
+                @title={{i18n "more_options"}}
+                @icon="ellipsis-vertical"
+                class="btn-small"
+              >
+                <:content>
+                  <DropdownMenu class="d-page-header__mobile-actions">
+                    {{#let
+                      (hash
+                        Primary=PrimaryActionListItem
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/d-page-header.gjs:66]: Yes good point, will change

File: app/assets/javascripts/discourse/app/components/d-page-action-button.gjs (hunk 3 above)
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/d-page-action-button.gjs:110]: Yes good point, will change

File: app/assets/javascripts/admin/addon/templates/config-look-and-feel.hbs (hunk 2 above)
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/templates/config-look-and-feel.hbs:7]: To me `@route` refers to an actual ember route whereas `@path` is just the end part of the URL. We did use a route initially for these, but I worked with  CvX on it and it was causing too many errors on page transitions so we went with a raw string route instead.

Would prefer to keep this as `@path` for now unless you feel strongly

File: app/assets/javascripts/admin/addon/templates/config-about.hbs (hunk 1 above)
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/templates/config-about.hbs:3]: Ok fair, I wasn't too happy with this double-up as well. Will change

File: app/assets/javascripts/admin/addon/templates/config-look-and-feel.hbs [hunk 5]
```diff
@@ -1,9 +1,10 @@
-<AdminPageHeader
+<DPageHeader
   @titleLabel="admin.config_areas.look_and_feel.title"
   @descriptionLabel="admin.config_areas.look_and_feel.description"
   @learnMoreUrl="https://meta.discourse.org/t/beginners-guide-to-using-discourse-themes/91966"
 >
   <:breadcrumbs>
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/templates/config-look-and-feel.hbs:6]: Yes fair, I do this for `actions` already so will make it consistent :+1:

[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/templates/config-look-and-feel.hbs:6]: Actually not sure how I can do this, at least not the `DPageHeader as |header|` part. In my case, I need to render these yields in different parts of the `DHeader` component. For FormKit's Form component app/assets/javascripts/discourse/app/form-kit/components/fk/form.gjs , you are able to `yield` one huge hash that renders components one after another into the container.

I could at least change `{{yield to="breadcrumbs"}}` to something like `{{yield (hash Item=DBreadcrumbItem}} to="breadcrumbs"}} and then this:

```handlebars
<:breadcrumbs as |breadcrumbs|>
  <breadcrumbs.Item @path="/admin" @label={{i18n "admin_title"}} />
  <breadcrumbs.Item
    @path="/admin/customize/emojis"
    @label={{i18n "admin.emoji.title"}}
  />
</:breadcrumbs>
```

And same for `:tabs/NavItem`. Am I missing something here, is there some way to do what you are suggesting and still render into named yields in different parts of the component?

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**